### PR TITLE
OSIDB-3275: Prevent flaw saving if CVSS vector is invalid

### PR DESCRIPTION
**Pull request by MrMarble**
# OSIDB-3275: Prevent flaw saving if CVSS vector is invalid

## Checklist:

- [x] Commits consolidated
- [x] Changelog updated
- [x] Test cases added/updated
- [x] Jira ticket updated

## Summary:

* Added CVSS validation to `validate` method in `useFlawModel` composable to prevent saving the flaw if the vector is invalid
* Renamed `useFlaws.spec.ts` to `useFlawsFetching.spec.ts`
* Moved fixtures to the top level folder for using them outside of the `components` folder
* Added tests to `useFlawModel`

## Considerations:

Closes OSIDB-3275

### CODE REVIEW

File: src/__tests__/__fixtures__/sampleFlawRequired.json [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
+    "cve_id": "CVE-2024-0001",
     "impact": "MODERATE",
     "components": [
         "kernel"
     ],
     "title": "Sample title",
     "trackers": [],
     "comment_zero": "Comment Zero == Patient Zero",
-    "cve_description": "",
+    "cve_description": "Some description",
```
Comment (C-Valen) [inline src/__tests__/__fixtures__/sampleFlawRequired.json:11]: Why was this added?

**Review by C-Valen — COMMENTED** [review]
Generally looks good. Weak preference to add small scope description to branches.

File: src/__tests__/__fixtures__/sampleFlawRequired.json (hunk 1 above)
Comment (MrMarble) [reply src/__tests__/__fixtures__/sampleFlawRequired.json:11]: the `requires_cve_description` field has the value `APPROVED`, and that only can happen if the description is not empty, another solution could be to change the value of the `requires_cve_description`

Comment (C-Valen) [reply src/__tests__/__fixtures__/sampleFlawRequired.json:11]: Oh good catch. I'd prefer that second approach, removing the requires state, just cause `cve_description` is not a required field.

Comment (superbuggy) [reply src/__tests__/__fixtures__/sampleFlawRequired.json:11]: If the real-world workflow is that a CVE scoring description is required, then maybe this good to add some example description

**Review by superbuggy — DISMISSED** [review]

File: src/composables/__tests__/useFlawModel.spec.ts [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+      const { flawRhCvss3, updateFlaw } = mountFlawModel(sampleFlawFull as ZodFlawType);
+
+      flawRhCvss3.value.vector = 'invalid';
+      await flushPromises();
+      updateFlaw();
+
+      expect(putFlaw).not.toHaveBeenCalled();
+    });
+  });
+});
```
Comment (superbuggy) [inline src/composables/__tests__/useFlawModel.spec.ts:124]: A welcome addition to our coverahe

**Review by C-Valen — APPROVED** [review]

**Pull request merged**